src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx

```diff
@@ -0,0 +1,276 @@
+import React,{useCallback,useState,useRef,useEffect} from 'react'
+import styled from 'styled-components'
+import {Modal as M} from '@Containers/Modal/Modal'
+import {Paragraph as P} from '@Text'
+import {MainTheme} from '@Themes'
+import { Main, MainInterface } from '@Utils/BaseStyles';
+import {Mixins} from '@Utils'
+import {Button} from '@Inputs/Button/Button'
+import {Input} from '@Inputs/Input/Input'
+import {Card} from '@Containers/Card/Card'
+
+const DECIMAL_BASE=10
+const TITLE_CONTAINER_TIMES_VERTICAL_PADDING=1.5
+const TITLE_CONTAINER_TIMES_HORIZONTAL_PADDING=3
+const TITLE_CONTAINER_OPACITY=0.8
+const ITEMS_CONTAINER_MIN_WIDTH=260
+const INPUT_WIDTH=30
+const RED_CARD_MIN_WIDTH=216
+const TIMES_BATCH_UPDATE_MODAL_PADDING=2
+const NOT_REDEEMABLE=0
+
+interface IReferenceValues{
+    setRedemptionPoints:React.Dispatch<React.SetStateAction<number>>;
+    redemptionPoints:number;
+}
+
+interface IModalProps{
+    state:[boolean,React.Dispatch<React.SetStateAction<boolean>>];
+}
+
+export interface IData{
+    name:string;
+    redemptionPoints:number;
+}
+
+export interface IItemsRedemptionPointsProps{
+    modalProps:IModalProps;
+    titleText:string;
+    titleDescription:string;
+    cancelButtonText:string;
+    applyButtonText:string;
+    batchUpdateButtonText:string;
+    onClickCancelButton:()=>void;
+    onClickApplyButton:(dataItems:IData[])=>void;
+    data:IData[];
+    applyToAllItemsText:string;
+}
+
+export const ItemsRedemptionPoints:React.FC<IItemsRedemptionPointsProps>=({modalProps,titleText,titleDescription,cancelButtonText,applyButtonText,batchUpdateButtonText,onClickApplyButton,onClickCancelButton,data,...props}):React.ReactElement=>{
+    const [isShownBatchUpdateModal,setIsShownBatchUpdateModal]=useState(false)
+
+    /**
+     * this is used to store info about the ItemRedemptionPoints child components
+     */
+    const infoRef=useRef<IReferenceValues[]>(new Array(data.length))
```

Comment:
Why not an array of values in state?

Reply:
I don't know, I am kind used to do it this way. I don't like too much to uplift state.

Reply:
You know when u use REF UI won't update, but when you update State UI will update right?

Reply:
Ref is not for data, it's for referencing components please fix this